src/renderer/hooks/useSessionStorage.ts

```diff
@@ -1,11 +1,19 @@
 import { useEffect, useState } from 'react';
 
+const safeJsonParse = (value: string) => {
+    try {
+        return JSON.parse(value) as unknown;
+    } catch (e) {
+        return undefined;
+    }
+};
+
 export const getSessionStorageOrDefault = <T>(key: string, defaultValue: T): T => {
     const stored = sessionStorage.getItem(key);
     if (!stored) {
         return defaultValue;
     }
-    return JSON.parse(stored) as T;
+    return safeJsonParse(stored) as T;
```

Comment:
How could there ever be invalid JSON in there?

Reply:
the string "undefined" apparently does not like being json parsed

Reply:
remove this and you'll crash when you reload (without having a file opened/saved)

Reply:
Well, that's because you are using `useSessionStorage` incorrectly. This hook only supports JSON values, and `undefined` is not a JSON value. Use `null` instead.

Reply:
I still think it would be good to wrap the json parse around this. If you ever somehow get an invalid state in your session, it'll just continue to crash without a way to fix it (which is very bad)

Reply:
With this "fix", the JSON parse won't crash anymore, yes. Now the first consumer of the returned value will crash because `undefined` is guaranteed to not be returned by the API. With this fix, the API lies.

Reply:
Same for `null`. The consumer controls the `T`, not `useSession`.

Reply:
@joeyballentine

Reply:
I'll get rid of it then